### PR TITLE
[SOFT-234] Merge codegen into firmware

### PR DESCRIPTION
**Pull request by raiyansayeed**
Taken at commit [774c256](https://github.com/uw-midsun/codegen-tooling-msxiv/commit/774c25626dee75959f1a93082f0f1ab2375fe858) of the codegen-tooling-msxiv repo

### CODE REVIEW

File: codegen/scripts/data.py
Comment (copperium) [inline codegen/scripts/data.py:15]: It looks like you or autopep8 moved the `import can_pb2` below the `sys.path.append` construction, which is causing `make codegen` to fail on my machine - can you move it back?

File: codegen/.gitignore
Comment (copperium) [inline codegen/.gitignore:1]: Shouldn't this .gitignore just be merged into the root .gitignore?

File: codegen/can_messages.asciipb
Comment (copperium) [inline codegen/can_messages.asciipb:2]: lol this is way outdated, maybe update this with a new wiki page or otherwise just remove this

File: libraries/codegen-tooling/inc/can_msg_defs.go
Comment (copperium) [inline libraries/codegen-tooling/inc/can_msg_defs.go:1]: This file and `can_msg_defs.ts` shouldn't be here, please delete

File: README.md
Comment (copperium) [inline README.md:137]: @jessm what do you think about ditching generating for Go entirely so we don't have to deal with this since we're not using it? Is it valuable to keep maintaining it just in case?

**Review by copperium — CHANGES_REQUESTED** [review]
Thanks for taking the initiative to do this Raiyan!

File: README.md
Comment (jessm) [reply README.md:137]: I'm all in favour of throwing it out. "Maintaining" is a pretty strong word in a student design team, I think the less code we keep the better off we are.

File: codegen/can_messages.asciipb
Comment (jessm) [reply codegen/can_messages.asciipb:2]: this file should probably be json lol, but that's another PR

**Review by jessm — COMMENTED** [review]
:bless:

File: README.md
Comment (raiyansayeed) [reply README.md:137]: Just to confirm: we're throwing out the Go toolchain entirely?

Comment (copperium) [reply README.md:137]: Protobuf generation is the only thing we're using the Go toolchain for right? I'd say just throw out the whole thing since we aren't using it and it's less to maintain. I'd also say to ditch the TypeScript generation since we're also not using that (unless @AveryChiu64 you have a use for it in CAN explorer/telemetry/driver display?).

**Comment by copperium** [comment]
Also FYI: before doing the final merge we should update the PR with the latest codegen-tooling-msxiv cut since there are still active PRs on it

File: Makefile
Comment (copperium) [inline Makefile:322]: MPXE should probably be updated to depend on this and use the new dbc file location, but that's for another PR (no action requested for this comment). @jessm

File: Makefile
Comment (copperium) [inline Makefile:359]: An issue I didn't foresee: this also runs all the MPXE integration tests since they match the `test_*.py` glob. Could you prune the MPXE integration test dir too?

Also, you shouldn't need the dependency on `codegen_test`.

File: Makefile
Comment (copperium) [inline Makefile:339]: This errors out for me because it can't find system_can.dbc - I think you either need to `cd $(CODEGEN_DIR)` here, or in the script use `codegen/system_can.dbc` as the dbc path.

File: codegen/mock_can_data.py
Comment (copperium) [inline codegen/mock_can_data.py:30]: IMO you should `raise` or `sys.exit(1)` at the end of this `except` block - otherwise you get confusing errors later on when the script tries to use `DB`.

File: projects/solar/test/test_sense_mcp3427.c
Comment (copperium) [inline projects/solar/test/test_sense_mcp3427.c:179]: I'm not really sure why you're making unrelated changes in power_selection and solar? Especially since these seem to be redundant. In general unrelated changes should be submitted as a separate PR - e.g. if we have to revert a commit it's better that it reverts one thing and not a bunch of unrelated changes that we have to put back.

**Review by copperium — CHANGES_REQUESTED** [review]

**Review by copperium — APPROVED** [review]
LGTM :+1:

**Comment by copperium** [comment]
Will merge once I get around to figuring out how to deprecate codegen-tooling-msxiv properly

**Pull request merged**
